zeebe/atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java

```diff
@@ -178,6 +178,25 @@ public ChannelPool getChannelPool() {
     return channelPool;
   }
 
+  private static MessagingConfig verifyHeartbeatConfig(final MessagingConfig config) {
+    if (config.getHeartbeatInterval().isZero() && !config.getHeartbeatTimeout().isZero()) {
+      // Setting both to zero essentially disables the heartbeat mechanism which is valid.
```

Comment:
If both should be set to zero to disable the mechanism, should we have the ! before the second condition?

Reply:
Yup, thanks :facepalm: